src/http/client.cc

```diff
@@ -390,7 +390,14 @@ future<> client::do_make_request(connection& con, request& req, reply_handler& h
             });
         }
 
-        return handle(rep, con.in(rep)).finally([reply = std::move(reply)] {});
+        return handle(rep, con.in(rep)).finally([reply = std::move(reply), &con] {
+            if (reply->content_length != reply->consumed_content) {
+                http_log.trace("content was not fully consumed, content length is {} but consumed only {}, will close the connection",
```

Comment:
As a followup.

Dropping a connection and establishing a new one can be more costly than reading the remaining bytes. Need some (AIish) logic here that prefers consuming the remaining bytes and dropping them on the floor instead of dropping the whole connection

Reply:
AI-ish, sure :) 128KiB it is